server/src/main/java/org/opensearch/threadpool/ThreadPool.java

```diff
@@ -234,6 +236,7 @@ public ThreadPool(
         final int halfProcMaxAt5 = halfAllocatedProcessorsMaxFive(allocatedProcessors);
         final int halfProcMaxAt10 = halfAllocatedProcessorsMaxTen(allocatedProcessors);
         final int genericThreadPoolMax = boundedBy(4 * allocatedProcessors, 128, 512);
+        final int snapshotDeletionPoolMax = boundedBy(4 * allocatedProcessors, 64, 256);
```

Comment:
should it be `halfProc` else it can overwhelm the cluster manager ?

Reply:
These are only going to perform IO operations. I have also tested out on lower instance types and it works pretty well. On higher instance types, the impact is negligible.

Reply:
have we tested with more number of shards as well ?